### PR TITLE
16 implement challenges details page

### PR DESCRIPTION
**Pull request by JahanviKathiriya**
new challenge details page

challenge type more aligned with backend

minor fixes on create challenges page

ongoing hydration issue with my get screen size file 0_0

### CODE REVIEW

File: src/app/(logged-in)/challenges/view/[id]/page.tsx [hunk 1]
```diff
@@ -0,0 +1,87 @@
...
+                <Flex>
+                    <Text className="mb-2 pr-2" as="b">File:</Text>
+                    <Text className="mb-2">will be added as a feature soon</Text>
+                </Flex>
+            </Box>
+        </div>
+
+    </Stack>
+    </>;
+}
```
Comment (nhanaa) [inline src/app/(logged-in)/challenges/view/[id]/page.tsx:87]: Can you split this into a separate component maybe?

File: src/types/challenges.types.ts [hunk 2]
```diff
@@ -1,19 +1,35 @@
...
+    challenge_id: string;
+    challenge_name: string,
+    category: string,
+    challenge_description: string,
     points: number,
-    hint: string
+    division: number[],
 }
 
 export const columns = [
```
Comment (nhanaa) [inline src/types/challenges.types.ts:29]: Can we make all types capitalized to be consistent?

**Review by nhanaa — COMMENTED** [review]
Just some minor things. Sorry if I'm too opiniated about this stuff lol. Rest looks good.

**Review by nhanaa — APPROVED** [review]

File: src/types/challenges.types.ts [hunk 3]
```diff
@@ -1,19 +1,35 @@
-export type challenges = {
-    id: number;
-    name: string,
+export type Hint = {
+    value: string,
+    cost: number
+};
+
+export type Challenges = {
+    challenge_id: string;
```
Comment (GoZaddy) [inline src/types/challenges.types.ts:7]: looks good but we're missing the creator_name field here. i have an incoming PR so i can just add it in mine

**Pull request merged**